### PR TITLE
Apply new design and introduce components

### PR DESCRIPTION
**Pull request by zharro**

### CODE REVIEW

File: src/js/popup/actions/timer.js [hunk 1]
```diff
@@ -50,15 +50,15 @@ export function isTimerActive(data) {
   return data.timer.startTime > 0;
 }
 
-export function onTimerClick(store, selectedProjectId, selectedMergeRequestId) {
-  const data = store.getState();
-  if (!isTimerLoaded(data)) return;
+export function onTimerToggled(store) {
```
Comment (matrohin) [inline src/js/popup/actions/timer.js]: Can you please add TODO to make it proper redux-thunk action. Just to not forget.

File: src/js/popup.js [hunk 2]
```diff
@@ -7,116 +7,114 @@ import "../css/lib/bootstrap.min.css";
...
 import { startTimerCounter } from "./popup/view/timerCounterView";
 import reducer from "./popup/reducers";
 
-const store = createStore(reducer, applyMiddleware(reduxThunk));
-
-function getProjectFromPath(path) {
-  let elems = path.split("/");
-  if (elems.length > 2) {
-    return elems[1] + "/" + elems[2];
+export default class Popup {
```
Comment (matrohin) [inline src/js/popup.js]: Why do we export this? Is it used somewhere else?

File: src/js/popup/view/timerCounterView.js [hunk 3]
```diff
@@ -7,7 +7,6 @@ import { calculateSpentSeconds } from "../actions/timer";
 
 const timer = new Timer();
 const timerCounterElement = document.getElementById("timerCounter");
-const timerCounterHelpElement = document.getElementById("timerCounterHelp");
```
Comment (matrohin) [inline src/js/popup/view/timerCounterView.js]: Why was it deleted? How it works? We use it below.

File: src/js/popup/view/timerHandlingComponent.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      this._onTimerToggled()
+    );
+
+    this._cancelTrackingElement = document.getElementById("cancelTracking");
+    this._cancelTrackingElement.addEventListener("click", () =>
+      this._onTimerCancelled()
+    );
+  }
+
+  updateView(isStaring) {
```
Comment (matrohin) [inline src/js/popup/view/timerHandlingComponent.js]: "staring"? Please change to isStarted or something like that.

File: src/js/popup/actions/timer.js [hunk 5]
```diff
@@ -50,15 +50,15 @@ export function isTimerActive(data) {
...
+  if (!isTimerLoaded(state)) return;
 
-  if (!isTimerActive(data)) {
-    store.dispatch(startTimer(selectedProjectId, selectedMergeRequestId));
+  if (!isTimerActive(state)) {
+    store.dispatch(startTimer(state.mrInfo.projectId, state.mrInfo.mrId));
     startTimerCounter(store);
   } else {
-    store.dispatch(stopTimer(data.timer, data.settings));
+    store.dispatch(stopTimer(state.timer, state.settings));
```
Comment (matrohin) [inline src/js/popup/actions/timer.js]: I get 'TypeError: state.settings is null' probably from here

File: src/js/popup/view/view.js [hunk 6]
```diff
@@ -1,82 +0,0 @@
...
-}
-
-function updateButtonView(started) {
-  const timerButtonIcon = document.getElementsByClassName(
-    "timer-toggler-icon"
-  )[0];
-
-  if (started) {
-    timerButtonIcon.classList.replace("fa-play-circle", "fa-stop-circle");
-    browser.browserAction.setIcon({ path: imgStop });
```
Comment (matrohin) [inline src/js/popup/view/view.js]: Please reapply these lines.

After the change icon is not changing when timer is started and tooltip is not changed either

File: src/js/popup/view/view.js [hunk 7]
```diff
@@ -1,82 +0,0 @@
...
-import { updateTimerCounter } from "./timerCounterView";
-
-import imgStart from "../../../img/start-gray-16.png";
-import imgStop from "../../../img/stop-red-16.png";
-
-const timerBtn = document.getElementById("timerToggler");
-
-function updateView(data) {
-  updateButtonView(isTimerActive(data));
-  updateInputsView(data.timer.projectId, data.timer.mergeRequestId);
```
Comment (matrohin) [inline src/js/popup/view/view.js]: Please fix this bug:
1. Open MR1;
2. Start tracking;
3. Go to MR2;
4. Open popup and see that MR2 is chosen but MR1 shall be.

File: src/js/popup/view/view.js (hunk 6 above)
Comment (zharro) [reply src/js/popup/view/view.js]: Good catch!

**Pull request merged**